src/Shared/Recruit.Vacancies.Client/Infrastructure/Repositories/MongoDbApplicationReviewRepository.cs

```diff
@@ -65,21 +65,25 @@ public Task UpdateAsync(ApplicationReview applicationReview)
 
         public async Task<List<T>> GetForEmployerAsync<T>(string employerAccountId)
         {
-            var filter = Builders<BsonDocument>.Filter.Eq(EmployerAccountId, employerAccountId);
-            var collection = GetCollection<BsonDocument>();
+            var filter = Builders<T>.Filter.Eq(EmployerAccountId, employerAccountId);

```

Comment:
Shouldn't this repository and the VacancyRepository return the full document domain object and mapping done in the service layer to the summary types. I thought Repository classes are for domain objects (read/write) and anything else would be Provider classes.

If we know it is a specific DTO would it not be more efficient to write the projection because if we are that concerned it would be a very specific use case.

Reply:
Maybe. I'm just trying to make the existing functionality a bit more throughput friendly.

Reply:
I agree with both of you :-)
I think this probably requires further discussion. I think what muddies the water is that we're using the repositories for queries not just when we're handling a create/update/delete and are getting an entity to update. 

@mgwilliam do you have any views on this?

Reply:
I'm inclined to separate the query and repository operations as there's a clear difference in intent:

* repository read/write methods are to support changing the data
* queries methods are for non-trivial reads that need to be current (ie. can't be stale)

... also noting that the query store is used for slow moving / complex reads or consolidating numerous data sources into a single doc/view.

So, given the above, I'd envisage the "projection methods" to move to the query interface from the repository interface.

Regarding how the query is implemented... has the mongo aggregation pipeline been considered? That would likely be more efficient as the hard work can be done at the data tier rather than pulling back into C#

Happy to discuss when we're all in the office if preferred?

Reply:
I'll split the repo interfaces into xxxRepository and xxxQuery.

I'll have a look if aggregates would be more appropriate for some of these query functions.

Reply:
In these cases there is nothing to be gained from using aggregates.